### PR TITLE
WIP: Feature/20 quarkus3 beta1

### PR DESCRIPTION
**Pull request by speter555**

### CODE REVIEW

**Comment by speter555** [comment]
[INFO] --- quarkus:3.0.0.Final:dev (default-cli) @ sample-quarkus-rest-service ---
[INFO] Invoking org.apache.maven.plugins:maven-resources-plugin:3.2.0:resources @ sample-quarkus-rest-service
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 5 resources
[INFO] Copying 1 resource
[INFO] The encoding used to copy filtered properties files have not been set. This means that the same encoding will be used to copy filtered properties files as when copying other filtered resources. This might not be what you want! Run your build with --debug to see which files might be affected. Read more at https://maven.apache.org/plugins/maven-resources-plugin/examples/filtering-properties-files.html
[INFO] Invoking io.quarkus:quarkus-maven-plugin:3.0.0.Final:generate-code @ sample-quarkus-rest-service
[INFO] Invoking org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile @ sample-quarkus-rest-service
[INFO] Nothing to compile - all classes are up to date
[INFO] Invoking org.apache.maven.plugins:maven-resources-plugin:3.2.0:testResources @ sample-quarkus-rest-service
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory /home/peterszabo/GIT/backend-sampler/sample-quarkus/sample-quarkus-rest-service/src/test/resources
[INFO] Invoking io.quarkus:quarkus-maven-plugin:3.0.0.Final:generate-code-tests @ sample-quarkus-rest-service
[INFO] Invoking org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile @ sample-quarkus-rest-service
[INFO] No sources to compile
Listening for transport dt_socket at address: 5005
Press [h] for more options>
Tests paused
Press [r] to resume testing, [h] for more options>
Press [r] to resume testing, [o] Toggle test output, [h] for more options>
2023-04-19 09:55:56.123 WARN  [thread:build-9] [io.quarkus.deployment.steps.ReflectiveHierarchyStep] [sid:] - Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
	- hu.icellmobilsoft.sampler.dto.sample.rest.post.SampleResponse (source: RestClientProcessor > hu.icellmobilsoft.sampler.dto.sample.rest.post.SampleResponse)
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.
2023-04-19 09:55:56.502 INFO  [thread:build-55] [io.quarkus.arc.processor.BeanProcessor] [sid:] - Found unrecommended usage of private members (use package-private instead) in application beans:
	- @Inject field hu.icellmobilsoft.coffee.rest.exception.DefaultExceptionMessageTranslator#baseApplicationContainer,
	- @Inject field hu.icellmobilsoft.coffee.rest.exception.DefaultExceptionMessageTranslator#localizedMessage,
	- @Inject field hu.icellmobilsoft.coffee.rest.exception.DefaultExceptionMessageTranslator#projectStage
	- and 2 more - please enable debug logging to see the full list
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2023-04-19 09:56:00.227 INFO  [thread:Quarkus Main Thread] [hu.icellmobilsoft.coffee.rest.validation.catalog.MavenURLStreamHandlerProvider] [sid:] - MavenURLStreamHandlerProvider registered
2023-04-19 09:56:02.441 INFO  [thread:Quarkus Main Thread] [io.quarkus] [sid:] - sample-quarkus-rest-service 2.0.0-SNAPSHOT on JVM (powered by Quarkus 3.0.0.Final) started in 10.717s. Listening on: http://localhost:8080
2023-04-19 09:56:02.448 INFO  [thread:Quarkus Main Thread] [io.quarkus] [sid:] - Profile dev activated. Live Coding activated.
2023-04-19 09:56:02.450 INFO  [thread:Quarkus Main Thread] [io.quarkus] [sid:] - Installed features: [cdi, config-yaml, jaeger, reactive-routes, rest-client, resteasy, resteasy-jaxb, resteasy-jsonb, security, servlet, smallrye-context-propagation, smallrye-fault-tolerance, smallrye-health, smallrye-jwt, smallrye-metrics, smallrye-openapi, smallrye-opentracing, swagger-ui, vertx]
2023-04-19 09:57:16.906 WARN  [thread:executor-thread-1] [hu.icellmobilsoft.sampler.common.rest.header.ProjectHeader] [sid:] - Request header doesnt contain (sessionToken) key

**Comment by speter555** [comment]
Plusz infó hogy a common-core-jee nem tartalmaz classokat így a quarkus:dev nem fut le emiatt.

File: sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+
+# Coffee configs
+coffee.app.name=${service.name}
+
+
+# Log format with extSessionId
+quarkus.log.console.format=%d{yyyy-MM-dd HH:mm:ss.SSS} %-5p [thread:%t] [%c{10}] [sid:%X{extSessionId}] - %s%E%n
+# Log All https://quarkus.io/guides/logging#quarkus-log-logging-log-config_quarkus.log.console.level
+# Available values: https://docs.jboss.org/jbossas/javadoc/7.1.2.Final/org/jboss/logmanager/Level.html
+quarkus.log.console.level=ALL
```
Comment (rombow) [inline sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties:16]: ALL nem tul sok? Csak a sajat logjainra vagyunk ALL szinten kivancsiak

File: sample-quarkus/sample-quarkus-rest-service/pom.xml [hunk 2]
```diff
@@ -10,62 +10,106 @@
 
 	<properties>
 		<service.name>quarkusrestsample</service.name>
-		<version.hu.icellmobilsoft.coffee>1.11.0</version.hu.icellmobilsoft.coffee>
-		<quarkus.microprofile>2.7.2.Final</quarkus.microprofile>
+		<version.hu.icellmobilsoft.coffee>2.0.0-SNAPSHOT</version.hu.icellmobilsoft.coffee>
```
Comment (rombow) [inline sample-quarkus/sample-quarkus-rest-service/pom.xml:13]: ezt valahogy ki kellene tudni szervezni egy helyre, ennyi helyen ki fogjuk felejteni fenttartani

Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/pom.xml:13]: Kitehetem gyökér pom.xml-be? Úgy ahogy a quarkus verziója van mert az kell a pluginhoz is, de az van a pom.xml-ben is használva, nem úgy mint ez lenne.

File: sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties (hunk 1 above)
Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties:16]: Beállítom úgy akkor ahogy szoktuk csinálni és INFO szinten lesz, de a mi logjaink TRACE-en.

File: sample-quarkus/sample-quarkus-rest-service/src/main/resources/META-INF/beans.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="https://jakarta.ee/xml/ns/jakartaee"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+      https://jakarta.ee/xml/ns/jakartaee
+      https://jakarta.ee/xml/ns/jakartaee/beans_4_0.xsd" bean-discovery-mode="annotated">
```
[the remark this answers is not in the document]
Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/src/main/resources/META-INF/beans.xml:6]: Igazából nincs is rá szüksége a quarkusnak, így ki is veszem

File: sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties (hunk 1 above)
Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties:16]: quarkus.log.console.level=INFO

quarkus.log.category."hu.icellmobilsoft".level=TRACE
quarkus.log.category."hu.icellmobilsoft.coffee.module.etcd.producer.DefaultEtcdConfigSource".level=INFO
quarkus.log.category."hu.hu.icellmobilsoft.coffee.module.etcd.service.EtcdService".level=INFO

**Review by rombow — DISMISSED** [review]

**Review by czenczl — DISMISSED** [review]

File: sample-quarkus/sample-quarkus-rest-service/src/main/resources/META-INF/beans.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/src/main/resources/META-INF/beans.xml]: Végülis kellett mégis, és levettem róla az annotated-es részt.

File: sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties (hunk 1 above)
Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/src/main/resources/application.properties]: Sikerült megtalálni, hogy megy. 
quarkus.log.level=INFO << minden log INFO szinten megy
quarkus.log.min-level=ALL << ha kell mélyebb szint, akkor ezzel a min-level-lel kell engedélyezni
és így lehet beállítani a category-kat:
quarkus.log.category."hu.icellmobilsoft".level=TRACE
quarkus.log.category."hu.icellmobilsoft.coffee.module.etcd.producer.DefaultEtcdConfigSource".level=INFO
quarkus.log.category."hu.hu.icellmobilsoft.coffee.module.etcd.service.EtcdService".level=INFO

File: sample-quarkus/sample-quarkus-rest-service/pom.xml [hunk 4]
```diff
@@ -10,62 +10,106 @@
...
 	</properties>
 
 	<dependencies>
+		<!-- Project -->
 		<dependency>
-			<groupId>io.quarkiverse.microprofile</groupId>
```
[the remark this answers is not in the document]
Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/pom.xml:19]: visszatettem és ennek következménye hogy most csak 3.0.0.Alpha3 van, mert abból még nem készült release.

File: sample-quarkus/sample-quarkus-rest-service/pom.xml (hunk 2 above)
Comment (speter555) [reply sample-quarkus/sample-quarkus-rest-service/pom.xml]: átrendezve. coffee kapott saját bom-ot, az van behúzva a quarkusba és az allba is.

**Review by bucherarnold — APPROVED** [review]

**Review by rombow — APPROVED** [review]

**Review by czenczl — DISMISSED** [review]

**Comment by speter555** [comment]
Mivel kijött a 3.0.0.Final, és már javító is, azt még szerintem jó lenne átvezetni

**Review by bucherarnold — APPROVED** [review]

**Review by vasizs — APPROVED** [review]

**Comment by speter555** [comment]
PR is cloesed because Quarkus sampler will be moved to other repo.

**Pull request closed without merging**